pom.xml

```diff
@@ -11,74 +11,181 @@
+                                            <!-- automatically creates the classpath using all project dependencies, also adding the project build directory -->
+                                            <classpath/>
+                                            <argument>-XX:+UnlockExperimentalVMOptions</argument>
+                                            <argument>-XX:+EnableJVMCI</argument>
+                                            <argument>--upgrade-module-path=${compiler.dir}/compiler.jar</argument>
+                                            <argument>com.mycompany.app.App</argument>
+                                        </arguments>
+                                    </configuration>
+                                </execution>
+                                <execution>
+                                    <id>nograal</id>
+                                    <goals>
+                                        <goal>exec</goal>
+                                    </goals>
+                                    <configuration>
+                                        <arguments>
+                                            <argument>--module-path</argument>
+                                            <!-- automatically creates the modulepath using all project dependencies, also adding the project build directory -->
+                                            <modulepath/>
+                                            <argument>-classpath</argument>
+                                            <!-- automatically creates the classpath using all project dependencies, also adding the project build directory -->
+                                            <classpath/>
+                                            <argument>com.mycompany.app.App</argument>
+                                        </arguments>
+                                    </configuration>
+                                </execution>
+                            </executions>
+                            <configuration>
+                                <executable>${JAVA_HOME}/bin/java</executable>
+                            </configuration>
+                        </plugin>
+                    </plugins>
+                </pluginManagement>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-dependency-plugin</artifactId>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
     <build>
         <pluginManagement>
             <plugins>
                 <plugin>
                     <artifactId>maven-compiler-plugin</artifactId>
                     <version>3.8.0</version>
                     <configuration>
-                        <source>11</source>
-                        <target>11</target>
+                        <source>1.8</source>
+                        <target>1.8</target>
```

Comment:
This definition is outside of profiles, right? E.g. then there should be no need to repeat it in the `profile`. It will be inherited.

Reply:
But the profile uses a different source and target versions.

Reply:
It does. But why? You are not using any fancy language features. Enough to stick to `1.8`, in my opinion.

Reply:
I'd argue that since this is a "starting point" project it should be clear which version we are using for what. And if someone changes the code to use java 11 features (var for example) it should be obvious that this will break GraalVM.